recipes/brpc/all/conanfile.py

```diff
@@ -0,0 +1,102 @@
+from conans import CMake, ConanFile, tools
+import glob
+import os
+
+
+class BrpcConan(ConanFile):
+    name = "brpc"
+    description = "An industrial-grade RPC framework used throughout Baidu"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apache/incubator-brpc"
+    topics = ("conan", "brpc", "baidu", "rpc")
+    license = ("Apache-2.0")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_snappy": [True, False],
+        "with_glog": [True, False],
+        "with_thrift": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "protobuf:with_zlib": True,
+        "leveldb:with_snappy": True,
```

Comment:
This is disallowed on cci for the same reasons as I explained in https://github.com/conan-io/conan-center-index/pull/2370#discussion_r465745218
This will probably result in 0 builds.
Maybe the default options of protobuf need to be revisited.
```suggestion
```

Reply:
Ok I removed this. I'm hoping this will be caught by the `ConanInvalidConfiguration` checks.

The reason I added this here, was based on my interpretation of the guidance In https://docs.conan.io/en/latest/mastering/conditional.html which states:

> Also, for options, a more flexible solution is to define dependencies options values in the default_options, not in the configure() method, as this would allow to override them.

Reply:
This should be catched in `build`.
The only problem now is that the ci will probably produce no builds.
As the default options of protobuf and leveldb are not compatible.